content/blog/2019/06/2019-06-25-phase-1-multibranch-pipeline-support-for-gitlab.adoc

```diff
@@ -0,0 +1,324 @@
+
+    [source, bash]
+    ---
+    mvn hpi:run # runs a Jenkins instance at localhost:8080
+    
+    or
+    
+    mvn hpi:run -Djetty.port={port} # to run on your desired port number 
+    ---
+
+    If you want to test it with your Jenkins server, after `mvn clean install` follow these steps in your Jenkins instance:
+    
+    1. Select `Manage Jenkins`
+    
+    2. Select `Manage Plugins`
+    
+    3. Select `Advanced` tab
+    
+    3. In `Upload Plugin` section, select `Choose file`
+    
+    4. Select `${root_dir}/target/gitlab-branch-source.hpi`
+    
+    5. Select `Upload` 
+    
+    6. Select `Install without restart`
+    
+== Usage
+
+Assuming plugin installation has done been already.
+
+=== Setting up GitLab Server Configuration on Jenkins
+
+1. On jenkins, select `Manage Jenkins`
+
+2. Select `Configure System`
+
+3. Scroll down to find the `GitLab` section
+
+   image::/images/post-images/2019-06-25-multibranch-pipeline-support-for-gitlab/gitlab-section.png[gitlab-section]
+
+4. Select `Add GitLab Server` | Select `GitLab Server`
+
+5. Now you will now see the GitLab Server Configuration options
+   
+   image::/images/post-images/2019-06-25-multibranch-pipeline-support-for-gitlab/gitlab-server.png[gitlab-server]
+   
+   There are 4 fields that needs to be configured:
+       
+    i. `Name` - Plugin automatically generates an unique server name for you. User may want to configure this field
+    to suit their needs but should make sure it is sufficiently unique. We recommend to keep it as it is.
+       
+    ii. `Server URL` - Contains the URL to your GitLab Server. By default it is set to "https://gitlab.com". User can
+    modify it to enter their GitLab Server URL e.g. https://gitlab.gnome.org/, http://gitlab.example.com:7990. etc.
+    
+    iii. `Credentials` - Contains a list of credentials entries that are of type GitLab Personal Access Token. When
+    no credential has been added it shows "-none-". User can add a credential by clicking "Add" button.
+    
+    iv. `Web Hook` - This field is a checkbox. If you want the plugin to setup a webhook on your GitLab project(s)
+    related jobs, check this box. The plugin listens to a URL for the concerned GitLab project(s) and when an event
+    occurs in the GitLab Server, the server sends an event trigger to the URL where the web hook is setup. If you
+    want continuous integration (or continuous delivery) on your GitLab project then you may want to automatically
+    set it up.
+     
+6. Adding a Personal Access Token Credentials:
+
+   This is a manual setup. To automatically generate Personal Access Token see
+   link:#creating-personal-access-token-within-jenkins[next section].
+
+    i. User is required to add a `GitLab Personal Access Token` type credentials entry to securely persist the token
+    inside Jenkins.
+
+    ii. Generate a `Personal Access Token` on your GitLab Server:
+        
+        a. Select profile dropdown menu from top-right corner
```

Comment:
```suggestion
... Select profile dropdown menu from top-right corner
```

Reply:
Images or additional lines in between auto-numbering breaks the chain. So moved images to the end of the section.